src/main/java/jenkins/branch/WorkspaceLocatorImpl.java

```diff
@@ -382,22 +389,27 @@ public static class Deleter extends ItemListener {
 
         private static /* almost final */ int CLEANUP_THREAD_LIMIT = SystemProperties.getInteger(Deleter.class.getName() + ".CLEANUP_THREAD_LIMIT", Integer.valueOf(0)).intValue();
 
+        private final transient ExecutorService executorService = Executors.newSingleThreadExecutor(threadFactory());
```

Comment:
I'm unsure; Will this survive a reload configuration from disk?

Reply:
I don't think this class is saved to disk at all; it's just an extension. If you restart Jenkins while a lot of deletion tasks are waiting to run, I do think they will all be lost if that's what you mean, but it looks like that was the case before this PR too.